### PR TITLE
init mining

### PR DESCRIPTION
**Pull request by syntrust**
Add hardhat scripts to deploy DecentrailizedKVDraggerHashimoto.sol and init shard to get ready for mining.

### CODE REVIEW

**Comment by qizhou** [comment]
what is the purpose of the PR

**Comment by qzhodl** [comment]
> what is the purpose of the PR

I think this PR wants to help to deploy DecentrailizedKVDraggerHashimoto.sol at 0x3330001 by calling the system deployer contract and init it. @syntrust you can provide more details.

@syntrust please notice that we may want every PR should provide a summary of PR like purpose, main change, how to test, etc.

**Comment by syntrust** [comment]
> > what is the purpose of the PR
> 
> I think this PR wants to help to deploy DecentrailizedKVDraggerHashimoto.sol at 0x3330001 by calling the system deployer contract and init it. @syntrust you can provide more details.
> 
> @syntrust please notice that we may want every PR should provide a summary of PR like purpose, main change, how to test, etc.

Yes. Added description.

**Comment by qzhodl** [comment]
> 

You may also want to add reviewers when creating a new PR, otherwise, people may not be notified by email.

**Review by qzhodl — APPROVED** [review]

File: scripts/initShard.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+const { ethers } = require("hardhat");
+
+async function main() {
+  const address = "0x0000000000000000000000000000000003330001"
+  const Contract = await ethers.getContractFactory('DecentralizedKVDaggerHashimoto');
+  const contract = Contract.attach(address);
+  const tx = await contract.initShard(0, "0x1111111111111111111111111111111111111111111111111111111111111111");
```
Comment (ping-ke) [inline scripts/initShard.js]: set to 1 instead of 0.

Comment (syntrust) [reply scripts/initShard.js]: done

File: package-lock.json
Comment (LiuJiazheng) [inline package-lock.json:1]: package-lock.json file is not necessary. Highly suggest to be removed

File: contracts/TestMerkleLib.sol [hunk 2]
```diff
@@ -34,7 +34,7 @@ contract TestMerkleLib {
         return MerkleLib.getProof(data, chunkSize, nChunkBits, chunkIdx);
     }
 
-    function getMaxLeafsNum(uint256 kvSize, uint256 chunkSize) public pure returns (uint256) {
-        return MerkleLib.getMaxLeafsNum(kvSize, chunkSize);
-    }
+    // function getMaxLeafsNum(uint256 kvSize, uint256 chunkSize) public pure returns (uint256) {
+    //     return MerkleLib.getMaxLeafsNum(kvSize, chunkSize);
+    // }
```
Comment (LiuJiazheng) [inline contracts/TestMerkleLib.sol:39]: Is there any particular reason to keep this function (even in comment)
Can it be removed entirely?

File: scripts/deploy.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+async function main() {
+  console.log("signer", (await ethers.getSigner()).address)
+  console.log("signer's balance", await (await ethers.getSigner()).getBalance())
+
+  const Contract = await ethers.getContractFactory('SystemDeployer');
+  const contract = await Contract.deploy();
+  await contract.deployed();
+  console.log('Contract deployed at ' + contract.address);
+  const addr = "0x0000000000000000000000000000000003330001"
```
Comment (LiuJiazheng) [inline scripts/deploy.js:11]: The string way is vulnerable. Especially many js scripts rely on this.
Perhaps having a common config file, and forcing everyone read from it, might be a better idea?

File: scripts/initShard.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+const { ethers } = require("hardhat");
+
+async function main() {
+  const address = "0x0000000000000000000000000000000003330001"
```
Comment (LiuJiazheng) [inline scripts/initShard.js:4]: ditto. Unreliable hand-written string.

File: scripts/initShard.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  const Contract = await ethers.getContractFactory('DecentralizedKVDaggerHashimoto');
+  const contract = Contract.attach(address);
+  const tx = await contract.initShard(1, "0x1111111111111111111111111111111111111111111111111111111111111111");
+  await tx.wait();
+  console.log("initShard success");
+  // const info = await contract.infos(0)
+  // console.log("info", info)
```
Comment (LiuJiazheng) [inline scripts/initShard.js:11]: It would be nice if, either we support a debug trace level, or remove them

**Pull request merged**